src/common/main.lua

```diff
@@ -1,29 +1,103 @@
 local json = require(".common.json")
 local utils = require(".common.utils")
 local main = {}
+-- just to ignore lint warnings
+local ao = ao or {}
 
-local camel = utils.camelCase
 main.init = function()
-	Owner = Owner or ao.env.Process.Owner
-	Name = Name or "ar.io ANT Registry"
+	ANT_REGISTRATION_TTL = 1000 * 60 * 30 -- 30 minutes
+	-- Example ANT structure
+	-- ANTS["antId"] = {
+	--     Owner = "userId",
+	--     Controllers = {"userId1" = true, "userId2" = true},
+	-- }
+	ANTS = ANTS or {}
+
+	-- Example ADDRESSES structure - keyed references to the above ANTS structure
+	-- ADDRESSES["userAddress"] = {"antProcessId1" = true, "antProcessId2" = true}
+	ADDRESSES = ADDRESSES or {}
 
 	local ActionMap = {
 		Register = "Register",
 		StateNotice = "State-Notice",
 		AccessControlList = "Access-Control-List",
 	}
 
-	Handlers.add("info", Handlers.utils.hasMatchingTag("Action", "Info"), function(msg)
+	utils.createActionHandler(ActionMap.Register, function(msg)
+		local antId = msg.Tags["Process-Id"]
+		assert(type(antId) == "string", "Process-Id tag is required")
+		assert(ANTS[antId] == nil, "ANT is already registered")
```

Comment:
do we care? if it's just passing along a `State` message than whats the penalty?